cmd/utils/cmd.go

```diff
@@ -293,6 +380,82 @@ func ExportAppendChain(blockchain *core.BlockChain, fn string, first uint64, las
 	return nil
 }
 
+// ExportHistory exports blockchain history into the specified directory,
+// following the Era format.
+func ExportHistory(bc *core.BlockChain, dir string, first, last, step uint64) error {
+	log.Info("Exporting blockchain history", "dir", dir)
+	if head := bc.CurrentBlock().Number.Uint64(); head < last {
+		log.Warn("Last block beyond head, setting last = head", "head", head, "last", last)
+		last = head
+	}
+	network := "unknown"
+	if name, ok := params.NetworkNames[bc.Config().ChainID.String()]; ok {
+		network = name
+	}
+	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
+		return fmt.Errorf("error creating output directory: %w", err)
+	}
+	var (
+		start     = time.Now()
+		reported  = time.Now()
+		checksums []string
+	)
+	for i := uint64(first); i <= last; i += step {
+		gen := func() error {
+			var (
+				buf = bytes.NewBuffer(nil)
+				w   = era.NewBuilder(buf)
+			)
+			for j := uint64(0); j < step && j <= last-i; j++ {
+				nr := i + j
```

Comment:
```suggestion
				number := i + j
```

Reply:
I think `n` or `nr` is okay here, `number` is rather verbose